crate/cli/src/actions/google/gmail_client/error.rs

```diff
@@ -0,0 +1,42 @@
+use thiserror::Error;
+
+/// Errors that may occur during any of the operations in this lib.
+#[derive(Error, Debug)]
+pub enum GoogleApiError {
+    /// A jwt error occurred.
+    #[error("JWT error `{0}`")]
+    JwtError(String),
+    /// Got an error whilst processing a request.
+    #[error("Reqwest error `{0}`")]
+    ReqwestError(String),
+    /// An IO error occurred.
+    #[error("Failed to load service account file `{0}`")]
+    ServiceAccountLoadFailure(String),
+    /// A serialization error occurred
+    #[error("Serialization error `{0}`")]
+    SerdeError(String),
+}
```

Comment:
You may want to keep errors unconverted:
```suggestion
pub enum GoogleApiError {
    /// A jwt error occurred.
    #[error("JWT error `{0}`")]
    JwtError(jwt_simple::Error),
    /// Got an error whilst processing a request.
    #[error("Reqwest error `{0}`")]
    ReqwestError(reqwest::Error),
    /// An IO error occurred.
    #[error("Failed to load service account file `{0}`")]
    IoError(io::Error),
    /// A serialization error occurred
    #[error("Serialization error `{0}`")]
    SerdeError(serde_json::Error),
}
```